tutorials/composite_bo_with_hogp.ipynb

```diff
@@ -0,0 +1,646 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Composite Bayesian Optimization with the Higher Order Gaussian Process"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this tutorial, we're going to explore composite Bayesian optimization [Astudillo & Frazier, ICML, '19](https://proceedings.mlr.press/v97/astudillo19a.html) with the Higher Order Gaussian Process (HOGP) model of [Zhe et al, AISTATS, '19](http://proceedings.mlr.press/v89/zhe19a.html). The setup for composite Bayesian optimization is that we have a function mapping input parameters to several outputs, and a second cheaper (and differentiable) function describing the quality of the functional output. We wish to find input parameters that maximize the output metric function in a black-box manner. \n",
+    "\n",
+    "Specifically, this can be described as $\\max_{x \\in \\mathcal{X}} g(f(x)),$ where $f: \\mathbb{R}^d \\rightarrow \\mathbb{R}^o$ and $g: \\mathbb{R}^o \\rightarrow \\mathbb{R}.$ As in traditional Bayesian optimization, we are going to construct a Gaussian process surrogate model over the expensive to evaluate function $f(.),$ and will use a HOGP to model this function. "
```

Comment:
```suggestion
    "Specifically, this can be described as $\\max_{x \\in \\mathcal{X}} g(f(x)),$ where $f: \\mathbb{R}^d \\rightarrow \\mathbb{R}^o$ and $g: \\mathbb{R}^o \\rightarrow \\mathbb{R}$, where $f$ is unknown and $g$ is known. As in traditional Bayesian optimization, we are going to construct a Gaussian process surrogate model over the expensive to evaluate function $f(.),$ and will use a HOGP to model this function. "
```